### PR TITLE
Define block icon as React component

### LINKED ISSUE #8401: Broken gradients in SVG icons due to duplicate IDs
## Bug Description

Original report. https://github.com/google/web-stories-wp/pull/8339#issuecomment-881460560

The icon for the web stories block is looks off in customizer widget screen in WordPress 5.8. See screenshot. 

## Steps to Reproduce

1. Open customizer.
2. Click on widgets.
3. Click on plus icon. 
4. Search web stories. 

## Screenshots

Screenshot 2021-07-16 at 14 42 28

## Additional Context

- Plugin Version: 1.9.0
- WordPress Version: 5.8
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by swissspidy**
## Context

The block icon could have had broken gradients, see #8401

## Summary

Moves the block icon to a React component to ensure unique IDs for each instance of the icon.

## User-facing changes

Screenshot 2021-07-23 at 14 25 36

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1. Go to Appearance -> Customizer to insert the Web Stories block a few times.
2. Ensure the block icon in the block toolbar is always visible and presented correctly.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #8401

### CODE REVIEW

**Comment by swissspidy** [comment]
@spacedmonkey This bug annoyed me so much I couldn't stop investigating after testing your fix. This is a possible workaround with a different approach.

**Comment by spacedmonkey** [comment]
Lol. It bugged me as well, which is why I looked into it. 

I like this solution, I think it is workable.

File: packages/stories-block/src/block/icons.js [hunk 1]
```diff
@@ -76,3 +81,58 @@ export const CIRCLE_CAROUSEL_CONFIG_ICON = (
 export const GRID_VIEW_CONFIG_ICON = (
   <GridViewConfigIcon width={20} height={27} />
 );
+
+// Defining the SVG like this ensures that IDs are unique
+// even if there are multiple instances of the component.
+// See https://github.com/google/web-stories-wp/issues/8401
+export const BlockIcon = () => {
+  const id = useInstanceId(BlockIcon);
+
```
Comment (spacedmonkey) [inline packages/stories-block/src/block/icons.js:90]:
```suggestion
const gradientId = `gradient-${id}`;
const filter          = `filter-${id}`l
```

Seems a little cleaner to setup these as consts up top.

**Review by spacedmonkey — APPROVED** [review]
Approved with Nitpic.

**Pull request merged**